app/models/principal.rb

```diff
@@ -86,8 +86,9 @@ def self.search_scope_without_project(project, query)
     not_locked.like(query).not_in_project(project)
   end
 
-  def self.order_by_name
-    order(User::USER_FORMATS_STRUCTURE[Setting.user_format].map { |format| "#{Principal.table_name}.#{format}" })
+  def self.order_by_name(desc: false)
+    OpenProject::Deprecation.warn "Use Queries::Principals with .order('name') instead"
+    order Queries::Principals::Orders::NameOrder.name_order_statements(desc)
```

Comment:
I am not yet decided on all of this and am very open for discussion. Currently I'd rather place more of the functionality in scopes again and use them in the query order objects. 

Scopes seem to be more easy to use throughout the code. This of course is highly subjective.

Reply:
I replaced it with a separate scope

Reply:
Merci